lib/typing/ext/time.go

```diff
@@ -51,29 +52,23 @@ type ExtendedTime struct {
 	nestedKind NestedKind
 }
 
+// MarshalJSON is a custom JSON marshaller for ExtendedTime.
 func (e ExtendedTime) MarshalJSON() ([]byte, error) {
-	return json.Marshal(e.String(""))
+	return json.Marshal(e.ts)
 }
 
+// TODO: Have this return an error instead of nil
 func NewExtendedTime(t time.Time, kindType ExtendedTimeKindType, originalFormat string) *ExtendedTime {
-	if originalFormat == "" {
-		switch kindType {
-		case TimestampTZKindType:
-			originalFormat = TimestampTZ.Format
-		case TimestampNTZKindType:
-			originalFormat = TimestampNTZ.Format
-		case DateKindType:
-			originalFormat = Date.Format
-		case TimeKindType:
-			originalFormat = Time.Format
-		}
+	defaultLayout, err := kindType.defaultLayout()
```

Comment:
Could we only do this if `originalFormat` is nil, so we don't hit an error if we're not going to use the default anyway?

Reply:
We can, I figure it's safer to error if the `kindType` is something we don't expect.

For example, some random caller passing in "foo" as a `ExtendedTimeKindType`, it's probably better to error when this function gets called rather than somewhere later in downstream